src/interpolation/details/ArborX_InterpDetailsMovingLeastSquaresCoefficients.hpp

```diff
@@ -26,13 +26,206 @@
+
+  template <typename T>
+  using UnmanagedView =
+      Kokkos::View<T, ScratchMemorySpace, Kokkos::MemoryUnmanaged>;
+
+  using LocalSourcePoints = Kokkos::Subview<SourcePoints, int, Kokkos::ALL_t>;
+  using LocalPhi = UnmanagedView<CoefficientsType *>;
+  using LocalVandermonde = UnmanagedView<CoefficientsType **>;
+  using LocalMoment = UnmanagedView<CoefficientsType **>;
+  using LocalSVDDiag = UnmanagedView<CoefficientsType *>;
+  using LocalSVDUnit = UnmanagedView<CoefficientsType **>;
+  using LocalCoefficients = Kokkos::Subview<Coefficients, int, Kokkos::ALL_t>;
+
+public:
+  MovingLeastSquaresCoefficientsKernel(ExecutionSpace const &,
+                                       TargetAccess const &target_access,
+                                       SourcePoints const &source_points,
+                                       Coefficients const &coefficients)
+      : _target_access(target_access)
+      , _source_points(source_points)
+      , _coefficients(coefficients)
+      , _num_targets(target_access.size())
+      , _num_neighbors(source_points.extent_int(1))
+  {}
+
+  std::size_t team_shmem_size(int const team_size) const
+  {
+    std::size_t val = 0;
+    val += LocalPhi::shmem_size(_num_neighbors);
+    val += LocalVandermonde::shmem_size(_num_neighbors, poly_size);
+    val += LocalMoment::shmem_size(poly_size, poly_size);
+    val += LocalSVDDiag::shmem_size(poly_size);
+    val += LocalSVDUnit::shmem_size(poly_size, poly_size);
+    return team_size * val;
+  }
+
+  template <typename TeamMember>
+  KOKKOS_FUNCTION void operator()(TeamMember member) const
+  {
+    int const rank = member.team_rank();
+    int const size = member.team_size();
+    auto const &scratch = member.thread_scratch(0);
+
+    int target = member.league_rank() * size + rank;
+    if (target >= _num_targets)
+      return;
+
+    auto target_point = _target_access(target);
+    auto source_points = Kokkos::subview(_source_points, target, Kokkos::ALL);
+    LocalPhi phi(scratch, _num_neighbors);
+    LocalVandermonde vandermonde(scratch, _num_neighbors, poly_size);
+    LocalMoment moment(scratch, poly_size, poly_size);
+    LocalSVDDiag svd_diag(scratch, poly_size);
+    LocalSVDUnit svd_unit(scratch, poly_size, poly_size);
+    auto coefficients = Kokkos::subview(_coefficients, target, Kokkos::ALL);
+
+    // The goal is to compute the following line vector for each target point:
+    // p(x).[P^T.PHI.P]^-1.P^T.PHI
+    // Where:
+    // - p(x) is the polynomial basis of point x (line vector).
+    // - P is the multidimensional Vandermonde matrix built from the source
+    //   points, i.e., each line is the polynomial basis of a source point.
+    // - PHI is the diagonal weight matrix / CRBF evaluated at each source
+    // point.
+
+    // We first change the origin of the evaluation to be at the target point.
+    // This lets us use p(0) which is [1 0 ... 0].
+    sourceRecenteringKernel(target_point, source_points);
```

Comment:
For me, a "kernel" is whatever is launched from the host, i.e., a `__global__` function in `Cuda`-speech.
```suggestion
    sourceRecentering(target_point, source_points);
```